src/components/CommunityEvents.tsx

```diff
@@ -0,0 +1,256 @@
+//Libraries
+import React, { useEffect, useState } from "react"
+import { useI18next, useTranslation } from "gatsby-plugin-react-i18next"
+import {
+  Box,
+  Center,
+  Divider,
+  Flex,
+  Heading,
+  Icon,
+  Text,
+} from "@chakra-ui/react"
+import { FaDiscord } from "react-icons/fa"
+import { DateTime } from "luxon"
+
+// Components
+import ButtonLink from "./ButtonLink"
+import Link from "./Link"
+import Translation from "./Translation"
+
+// Constants
+import { GATSBY_FUNCTIONS_PATH } from "../constants"
+
+// Utils
+import { getData } from "../utils/cache"
+
+interface Event {
+  date: string
+  title: string
+  calendarLink: string
+  pastEventLink: string | undefined
+}
+
+interface State {
+  pastEventData: Array<Event>
+  upcomingEventData: Array<Event>
+  loading: boolean
+  hasError: boolean
+}
+
+const DiscordButton = () => {
+  return (
+    <ButtonLink to={"/discord/"} gap={2}>
+      <Icon as={FaDiscord} fontSize={25} />
+      Join Discord
+    </ButtonLink>
+  )
+}
+
+const CommunityEvents = () => {
+  const { language } = useI18next()
+  const { t } = useTranslation()
+  const [state, setState] = useState<State>({
+    pastEventData: [],
+    upcomingEventData: [],
+    loading: true,
+    hasError: false,
+  })
+
+  useEffect(() => {
+    try {
+      const fetchCalendarData = async () => {
+        const events = await getData(`${GATSBY_FUNCTIONS_PATH}/calendarEvents`)
+        const pastEventData = events.pastEvents.map((event) => {
+          return {
+            date: event.start.dateTime,
+            title: event.summary,
+            calendarLink: event.htmlLink,
+            pastEventLink: event.location,
+          }
+        })
+        const upcomingEventData = events.futureEvents.map((event) => {
+          return {
+            date: event.start.dateTime,
+            title: event.summary,
+            calendarLink: event.htmlLink,
+            pastEventLink: event.location,
+          }
+        })
+        setState({
+          ...state,
+          pastEventData,
+          upcomingEventData,
+          loading: false,
+          hasError: false,
+        })
+      }
+      fetchCalendarData()
+    } catch {
+      setState({ ...state, loading: false, hasError: true })
+    }
+  }, [])
+
+  const renderEventDateTime = (
+    date,
+    language,
+    params = {
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+      hour12: false,
+      hour: "numeric",
+      minute: "numeric",
+    }
+  ) => {
+    return DateTime.fromISO(date).setLocale(language).toLocaleString(params)
+  }
+
+  const renderEventLink = (link, title) => {
```

Comment:
Since this is rendering jsx, lets create a component for it.

Also, I don't think we really need to define it as we are not abstracting too many things (just the `fontWeight`). However, if you want to keep it, I'd recommend doing the following:
```tsx
const EventLink = (props) => (
  <Link fontWeight="700" {...props} />
)

// usage
<EventLink to={..}>{title}</EventLink>
```